Mage.Sets/src/mage/cards/a/AKillerAmongUs.java

```diff
@@ -0,0 +1,236 @@
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+        Permanent permanent = game.getPermanentOrLKIBattlefield(source.getSourceId());
+        if (permanent == null) {
+            return false;
+        }
+
+        Choice choice = new ChoiceImpl();
+        Set<String> choices = new LinkedHashSet<>();
+        choices.add("Human");
+        choices.add("Merfolk");
+        choices.add("Goblin");
+        choice.setChoices(choices);
+        choice.setMessage("Choose Human, Merfolk, or Goblin");
+
+        controller.choose(outcome, choice, game);
+        game.informPlayers(permanent.getName() + ": " + controller.getLogName() + " has secretly chosen a creature type.");
+
+        SubType chosenType = SubType.fromString(choice.getChoice());
+        setSecretCreatureType(chosenType, source, game);
+        setSecretOwner(source.getControllerId(), source, game);
+        return true;
+    }
+
+    public static void setSecretCreatureType(SubType type, Ability source, Game game) {
+        game.getState().setValue(getThing(source, game) + SECRET_CREATURE_TYPE, type);
+    }
+
+    public static SubType getSecretCreatureType(Ability source, Game game) {
+        return (SubType) game.getState().getValue(getThing(source, game) +
+                ChooseHumanMerfolkOrGoblinEffect.SECRET_CREATURE_TYPE);
+    }
+
+    public static void setSecretOwner(UUID owner, Ability source, Game game) {
+        game.getState().setValue(getThing(source, game) + SECRET_OWNER, owner);
+    }
+
+    public static UUID getSecretOwner(Ability source, Game game) {
+        return (UUID) game.getState().getValue(getThing(source, game) +
+                ChooseHumanMerfolkOrGoblinEffect.SECRET_OWNER);
+    }
+
+    private static String getThing(Ability source, Game game) {
+        if (game.getPermanentOrLKIBattlefield(source.getSourceId()) != null) {
+            return "" + source.getSourceId() + "_" + (game.getPermanentOrLKIBattlefield(source.getSourceId()).getZoneChangeCounter(game));
+        }
+        return "Does not exist";
+    }
+}
+
+class AKillerAmongUsEffect extends OneShotEffect {
+
+    AKillerAmongUsEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "If target attacking creature token is the chosen type, " +
+                "put three +1/+1 counters on it and it gains deathtouch until end of turn.";
+    }
+
+    private AKillerAmongUsEffect(final AKillerAmongUsEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public AKillerAmongUsEffect copy() {
+        return new AKillerAmongUsEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent creature = game.getPermanent(source.getFirstTarget());
+        if (creature == null) {
+            return false;
+        }
+        SubType creatureType = ChooseHumanMerfolkOrGoblinEffect.getSecretCreatureType(source, game);
```

Comment:
I'm not sure if it's possible for `creature.getSubType()` to end up containing `null`, it shouldn't be, but probably best to null check this anyway